### DISCUSSION TITLE
Add "transfer ownership" and "make a copy" features to all tools

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5561] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
From survey results and past issues, I see that users want to transfer ownership of Queries, Reports and Record Sets. Making a duplicate would be nice too.

**Describe the solution you'd like**
Add a "transfer ownership" and "make a copy" feature for Queries, Reports and Record Sets

**Describe alternatives you've considered**
Editing the database directly

**Reported By**
Many

**Comment by grantfitzsimmons** [comment]
The ability to have a "default" query (or queries) that is included for every user in a collection/division/institution should be an option as well. If this could be configured to be global and/or non-editable, that would be even better.


**Requested by:** CSIRO

**Comment by maxpatiiuk** [comment]
> The ability to have a "default" query (or queries) that is included for every user in a collection/division/institution should be an option as well. If this could be configured to be global and/or non-editable, that would be even better.
> 
> **Requested by:** CSIRO

Related feature request: https://github.com/specify/specify7/issues/524

**Comment by grantfitzsimmons** [comment]
We should definitely have the ability to "lock" a query at that whichever level to not allow edits.

**Comment by grantfitzsimmons** [comment]
I want to add a +1 to this request. Being able to copy a query from one user to another would be extremely helpful and has been requested many times.

I would like to be able to copy a query mapping and paste it into a different query, potentially on a different user in the same DB for testing or troubleshooting (very similar to the WorkBench mapper mapping ability).

**Comment by maxpatiiuk** [comment]
Does query import/export work for you?
See https://github.com/specify/specify7/issues/1220

**Comment by grantfitzsimmons** [comment]
That half of the function is working fine, thanks! Still adding to another user would be nice 👍

**Comment by maxpatiiuk** [comment]
This feature should probably be implemented on the front end (as it stores the localization strings for the notification messages), but there is no way for the front end to send a notification to another user.

@benanhalt Need to implement the following first: https://github.com/specify/specify7/issues/1816. Until then, this is blocked.

---

**Edit:**
as per https://github.com/specify/specify7/issues/1816#issuecomment-1224930602, I will try to implement this feature without notification sharing

**Comment by grantfitzsimmons** [comment]
Eawag would benefit from being able to share record sets.

**Comment by maxpatiiuk** [comment]
>  Eawag would benefit from being able to share record sets.

This likely requires a schema change to add a boolean field indicating whether the record set should be shared or not

Until then, opening the record set directly by ID is the only option

**Comment by grantfitzsimmons** [comment]
I've added the ability to transfer ownership of queries and record sets in https://github.com/specify/specify7/pull/5367
